frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -1475,7 +1476,7 @@ const TypedFnSignature* instantiateSignature(Context* context,
 
   ResolutionResultByPostorderID r;
   auto visitor = createResolverForFnOrAd(context, fn, ad, substitutions,
-                                         poiScope, r);
+                                         poiScope, r, call.calledType());
```

Comment:
I'm not so sure about this one, but if you are more confident than I, it doesn't need to hold up the PR and a explainer or TODO comment would suffice.

Here, my understanding is that `call.calledType` is the type from the call site. But, that doesn't necessarily match the type of receiver for the method (due to implicit conversion or possibly instantiation). In the case of a generic method receiver in the method, we should be computing the method receiver type / "in composite type" from the instantiation's substitutions. In the case of a concrete method where an implicit conversion is in play (e.g. `proc int.mymethod() { } (1:int(8)).mymethod();`) then we should be getting the receiver type / "in composite type" information from the method declaration / initial signature.

Reply:
Good point. Should we add a method that sets the ``inCompositeType`` property once ``this`` has been instantiated (which would be in the loop below)?

Reply:
That sounds like a good solution for now, thanks